pkg/kubernetes/cluster.go

```diff
@@ -196,13 +230,20 @@ func setClusterCNI(config map[string]any) {
 	config[cniKey] = cniDefaultValue
 }
 
-func setClusterAPIAddress(config map[string]any, apiAddress string, port int) {
-	if apiAddress == "" {
+func setClusterAPIAddress(config map[string]any, ip4 *netip.Addr, ip6 *netip.Addr, port uint16, prioritizeIPv6 bool) {
+	if ip4 == nil && ip6 == nil {
 		zap.S().Warn("Attempted to set an empty cluster API address")
 		return
 	}
 
-	config[serverKey] = fmt.Sprintf("https://%s:%d", apiAddress, port)
+	switch {
+	case ip6 != nil && prioritizeIPv6:
+		config[serverKey] = fmt.Sprintf("https://%s", netip.AddrPortFrom(*ip6, port).String())
+	case ip6 != nil && ip4 == nil:
+		config[serverKey] = fmt.Sprintf("https://%s", netip.AddrPortFrom(*ip6, port).String())
+	default:
+		config[serverKey] = fmt.Sprintf("https://%s", netip.AddrPortFrom(*ip4, port).String())
+	}
```

Comment:
If I had to improve, I'd go with this. But I'm not convinced it should be part of this function. The code is the same for both so the checks could be extracted out of this and only a single address could be passed.

```suggestion
	if ip6 != nil && (prioritizeIPv6 || ip4 == nil) {
		config[serverKey] = fmt.Sprintf("https://%s", netip.AddrPortFrom(*ip6, port).String())
		return
	}

	config[serverKey] = fmt.Sprintf("https://%s", netip.AddrPortFrom(*ip4, port).String())
```

Reply:
Let's do something about this. The whole `ipv4Holder` / `ipv6Holder` pointer shenanigans also doesn't seem worth it to me.